controller/replica_controller.go

```diff
@@ -98,7 +99,8 @@ func NewReplicaController(
 
 		ds: ds,
 
-		rebuildingLock: &sync.Mutex{},
+		rebuildingLock:          &sync.Mutex{},
+		inProgressRebuildingMap: map[string]struct{}{},
```

Comment:
NIT: I would probably use `make(map[string]ValueType)` instead. The inline initialization is nice if you immediately add values. But again not important :)